test/sharness/t0080-repo.sh

```diff
@@ -98,7 +104,7 @@ test_expect_success "remove recursive pin, add direct" '
 
 test_expect_success "remove direct pin" '
 	echo "unpinned $HASH" >expected6 &&
-	ipfs pin rm "$HASH" >actual6 &&
+	ipfs pin rm -r=false "$HASH" >actual6 &&
```

Comment:
I agree that `rm` is a bad model.  The type of a pin does not say anything about the type of a file. In addition I image most pins are recursive and direct pins are only used in special cases.